common/address.go

```diff
@@ -41,3 +44,37 @@ func (addr Address) IsEmpty() bool {
 func (addr Address) String() string {
 	return string(addr)
 }
+
+func ConvertRecoverToError(r interface{}) error {
+	switch x := r.(type) {
+	case string:
+		return errors.New(x)
+	case error:
+		return x
+	default:
+		return errors.New(fmt.Sprint("%v", x))
```

Comment:
```suggestion
		return fmt.Errorf("%v", x)
```
Actually better
Might also be related to the lint error